components/romio/src/ome/io/nio/SimpleBackOff.java

```diff
@@ -84,7 +90,11 @@ public void throwMissingPyramidException(String msg, Pixels pixels) {
     }
 
     protected long calculate(Pixels pixels) {
-        return (long) (scalingFactor * countTiles(pixels));
+        // only count tiles if pixels report reasonable size values
+        return (pixels.getSizeC() > maxPixels || pixels.getSizeT() > maxPixels
+                || pixels.getSizeX() > maxPixels || pixels.getSizeY() > maxPixels || pixels
+                .getSizeZ() > maxPixels) ? defaultValue
+                : (long) (scalingFactor * countTiles(pixels));
```

Comment:
Does this `long` cast have to be *within* the following parentheses to have useful effect?

Reply:
Yes, some testing confirms that where it is it's essentially useless.

Reply:
`scalingFactor` is a double, so the result of the multiplication has to be cast to a long, if not it'll cause a compile error.

Reply:
Ah! Thank you, that makes more sense, sorry for the noise.

Reply:
No problem. I know it's a bit messy, double by int multiplation, casting to long... but it compiles ;-)